xbmc/filesystem/File.cpp

```diff
@@ -283,7 +288,10 @@ bool CFile::Open(const CURL& file, const unsigned int flags)
         return m_pFile->Open(url);
       }
     }
-    m_pFile = CFileFactory::CreateLoader(url);
+    if (m_pFile)
+      m_pFile->Close();
+    else
+      m_pFile = CFileFactory::CreateLoader(url);
```

Comment:
It's for every file. If you call open multiple times without closing close() before, the m_pFile pointer gets lost as memoryleak.